src/ioctls/vcpu.rs

```diff
@@ -233,6 +235,75 @@ impl VcpuFd {
         Ok(())
     }
 
+    /// Specifies set of signals that are blocked during execution of KVM_RUN.
```

Comment:
I believe this functionality is needed for older kernel versions that do not support KVM_IMMEDIATE_EXIT to be able to kick vcpus. I am not super sure we should add this as we currently "support" only newer kernel versions. By support here I mean that we are always using KVM bindings coming from a kernel version > 4.20 where KVM_IMMEDIATE_EXIT is available.

CC: @rust-vmm/gatekeepers are there any usecases for which we also want to support this?

Reply:
I agree, KVM_CAP_IMMEDIATE_EXIT should be enough; if we want to support both, I'd rather have higher-level functionality that on one hand sets up the signal handler properly for KVM_CAP_IMMEDIATE_EXIT, and on the other hand knows how to "eat" the signal with sigwaitinfo (after running KVM_RUN) when not using KVM_CAP_IMMEDIATE_EXIT.

Reply:
@andreeaflorescu , @bonzini . Thanks for reviewing.

Truely, `KVM_IMMEDIATE_EXIT` replaces `KVM_SET_SIGNAL_MASK` ioctl, there is good explanation in this page: http://patchwork.ozlabs.org/patch/726459 .

I have one thing to confirm about the last comment:
`KVM_SET_SIGNAL_MASK` ioctl is not needed even if KVM_CAP_IMMEDIATE_EXIT extension is not supported, right?

I will close this PR after getting confirmed.